### PR TITLE
feat(trace-view-load-more): Added parallelization of nodestore fetches.

### PR DESCRIPTION
**Pull request by Abdkhan14**
For task: [Workaround-for-the-100-Transaction-Limit-on-Trace-View](https://www.notion.so/sentry/Workaround-for-the-100-Transaction-Limit-on-Trace-View-b51eb23edf254dc8804f4a395df2147d)

This PR attempts to parallelize nodestore fetches made to build the trace view, under the feature flag `organizations:trace-view-load-more`.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events_trace.py [hunk 1]
```diff
@@ -793,6 +796,26 @@ def update_children(event: TraceEvent, limit: int) -> None:
                 child.generation = parent.generation + 1 if parent.generation is not None else None
                 parents.append(child)
 
+    # Concurrently fetches nodestore data to construct and return a dict mapping eventid of a txn
+    # to the associated nodestore event.
+    @staticmethod
+    def nodestore_event_map(events: Sequence[SnubaTransaction]) -> Dict[str, Optional[Event]]:
+        map = {}
+        with ThreadPoolExecutor(max_workers=20) as executor:
```
Comment (narsaynorath) [inline src/sentry/api/endpoints/organization_events_trace.py:804]: This should be feature flagged, right?

Comment (narsaynorath) [reply src/sentry/api/endpoints/organization_events_trace.py:804]: I also don't really see us going above 10 in the backend, maybe we should use that limit here as well?

**Review by narsaynorath — COMMENTED** [review]
lgtm, just some comments on feature flags and the # of workers

**Review by narsaynorath — APPROVED** [review]

File: src/sentry/api/endpoints/organization_events_trace.py (hunk 1 above)
Comment (Abdkhan14) [reply src/sentry/api/endpoints/organization_events_trace.py:804]: Yep you're right I'll feature flag it

**Pull request merged**